packages/expo/android/src/main/java/expo/modules/ReactActivityDelegateWrapper.kt

```diff
@@ -55,22 +69,39 @@ class ReactActivityDelegateWrapper(
   }
 
   override fun onCreate(savedInstanceState: Bundle?) {
-    // Since we just wrap `ReactActivityDelegate` but not inherit it, in its `onCreate`,
-    // the calls to `createRootView()` or `getMainComponentName()` have no chances to be our wrapped methods.
-    // Instead we intercept `ReactActivityDelegate.onCreate` and replace the `mReactDelegate` with our version.
-    // That's not ideal but works.
-    val reactDelegate = object : ReactDelegate(
-      plainActivity, reactNativeHost, mainComponentName, launchOptions
-    ) {
-      override fun createRootView(): ReactRootView {
-        return this@ReactActivityDelegateWrapper.createRootView()
+    // Give handlers a chance as early as possible to replace the wrapped delegate object.
+    // If they do, we call the new wrapped delegate's `onCreate` instead of overriding it here.
+    val newDelegate = reactActivityHandlers.asSequence()
+      .mapNotNull { it.onDidCreateReactActivityDelegate(activity, this) }
+      .firstOrNull()
+    if (newDelegate != null && newDelegate != this) {
+      val mDelegateField = ReactActivity::class.java.getDeclaredField("mDelegate")
```

Comment:
[proguard rules](https://github.com/expo/expo/blob/main/packages/expo/android/proguard-rules.pro) should be updated for accessing `ReactActivity.mDelegate`